### PR TITLE
Move `ServletContextPath` to `servlet-common:bootstrap`

### LINKED ISSUE #4317: Move io.opentelemetry.instrumentation.api.servlet to servlet-common
`io.opentelemetry.instrumentation.api.servlet` package from `instrumentation-api` module is used only by servlets and some methods from it are used only from `servlet-common` module. We can move that package closer to its usage and hide those methods from general public, who should not be concerned with them.

Thoughts? I can do it if get enough support.

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves #4317

Changes include:
* A couple of `jaxws` "library" instrumentations (which weren't _real_ library instrumentations) were removed and their code moved into their respective javaagent module
* Renamed `jsf-common:library` to `jsf-common:javaagent`
* Restlet library instrumentations are not using `ServletContextPath` anymore (there's no servlet library instrumentation so it makes no sense to use it there anyway)

`instrumentation-api` still contains `ServerSpanNaming`, but this particular class should perhaps be refactored into a general `http.route` utility in #442

### CODE REVIEW

File: instrumentation/jaxws/jaxws-2.0-axis2-1.6/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -17,14 +17,15 @@ muzzle {
 }
 
 configurations.configureEach {
-  if (!name.contains("muzzle")) {
-    // axis has a dependency on servlet api, get rid of it
+  if (name.contains("test")) {
```
Comment (trask) [inline instrumentation/jaxws/jaxws-2.0-axis2-1.6/javaagent/build.gradle.kts:20]: 👍

File: instrumentation/restlet/restlet-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v1_0/RestletSingletons.java [hunk 2]
```diff
@@ -16,9 +18,22 @@
   private static final Instrumenter<Request, Response> INSTRUMENTER =
       RestletTracing.create(GlobalOpenTelemetry.get()).getServerInstrumenter();
 
+  private static final ServerSpanNameSupplier<String> SERVER_SPAN_NAME =
+      (context, pattern) -> {
+        if (pattern == null || pattern.equals("")) {
+          return null;
+        }
+
```
Comment (trask) [inline instrumentation/restlet/restlet-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v1_0/RestletSingletons.java:26]: is this needed? ServletContextPath.prepend handles this case (and still prepends servlet context root)
```suggestion
```

File: instrumentation/restlet/restlet-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v2_0/RestletSingletons.java [hunk 3]
```diff
@@ -16,9 +18,22 @@
   private static final Instrumenter<Request, Response> INSTRUMENTER =
       RestletInstrumenterFactory.newServerInstrumenter(GlobalOpenTelemetry.get());
 
+  private static final ServerSpanNameSupplier<String> SERVER_SPAN_NAME =
+      (context, pattern) -> {
+        if (pattern == null || pattern.equals("")) {
+          return null;
+        }
+
```
Comment (trask) [inline instrumentation/restlet/restlet-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v2_0/RestletSingletons.java:26]: same

File: instrumentation/spring/spring-webmvc-3.1/javaagent/build.gradle.kts [hunk 4]
```diff
@@ -20,6 +20,8 @@ muzzle {
 val versions: Map<String, String> by project
 
 dependencies {
+  bootstrap(project(":instrumentation:servlet:servlet-common:bootstrap"))
```
Comment (trask) [inline instrumentation/spring/spring-webmvc-3.1/javaagent/build.gradle.kts:23]: it's nice to see/understand these dependencies better now

**Review by trask — APPROVED** [review]

File: instrumentation/restlet/restlet-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v1_0/RestletSingletons.java (hunk 2 above)
Comment (mateuszrzeszutek) [reply instrumentation/restlet/restlet-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/restlet/v1_0/RestletSingletons.java:26]: Good point -- I'll simplify that.

**Pull request merged**